OBA 2.x

aaronbrethorst opened the pull request:
## Changes

* Pull down OBA WAR and JAR files from Cambridge Systematics's Maven artifacts server
* Swap in MySQL for PostgreSQL
* Temporarily hardcode GTFS data within the Dockerfile
* Update data-sources.xml files to account for the latest changes in Hibernate, new DB server, etc.
* Change the exposed Tomcat port from `8888` to `8080`
* Update the docker-compose.yml file to account for these changes

## TODO

This branch isn't fully functional yet! See the [accompanying issue](https://github.com/OneBusAway/onebusaway-docker/issues/24) for a list of open items to fix.

CesarAndrade007 committed:
Modified docker-compose file to include JDBC and MySQL. Dockerfile Modified with updated images of maven and tomcat in addition to getting dependencies

CesarAndrade007 committed:
Updated Dockerfile

CesarAndrade007 committed:
Updated Dockerfile to include up-to-date dependencies

aaronbrethorst committed:
WIP: Rebuilds project to work with OBA Server 2.1.18-cs

* Pull down OBA WAR and JAR files from Cambridge Systematics's Maven artifacts server
* Swap in MariaDB (MySQL variant) for PostgreSQL
* Temporarily hardcode GTFS data within the Dockerfile
* Update data-sources.xml files to account for the latest changes in Hibernate, new DB server, etc.
* Change the exposed Tomcat port from `8888` to `8080`
* Update the docker-compose.yml file to account for these changes

TODO:

This branch isn't fully functional yet! Here's a list (likely incomplete) of tasks:

- [ ] The only endpoint I can get to respond is http://localhost:8080/onebusaway-api-webapp/api/where/current-time.json?key=test - anything that touches bundle data is unresponsive. Fix this.
- [ ] Update the bundle build process and move it back into a separate container in docker-compose.yml, like it was before
- [ ] Remove all Travis CI artifacts and switch over to GitHub Actions
- [ ] Go through the `acta-webapp` data sources XML file and validate it's correct
- [ ] Go through the `data-federation-webapp` data sources XML file and validate it's correct
- [ ] Go through the `api-webapp` data sources XML file and validate it's correct

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/OneBusAway/onebusaway-docker?pullRequest=25) <br/>All committers have signed the CLA.

ethanpooley commented on oba_config/onebusaway-transit-data-federation-webapp-data-sources.xml:
```diff
@@ -0,0 +1,38 @@
+    <property name="order" value="-2" />
+    <property name="properties">
+      <props>
+        <prop key="bundlePath">/var/lib/obanyc/no-such-dir</prop>
```
I'm seeing the bundle get built into `/oba/gtfs`, which is the path I would expect to see here.

ethanpooley commented on oba_config/onebusaway-transit-data-federation-webapp-data-sources.xml:
```diff
@@ -0,0 +1,38 @@
+  </bean>
+
+  <bean id="bundleManagementService" class="org.onebusaway.transit_data_federation.impl.bundle.BundleManagementServiceImpl">
+    <property name="bundleStoreRoot" value="/bundle" />
```
Same thing here. I'm seeing the bundle get built into `/oba/gtfs`, which is the path I would expect to see. When I changed these two paths, some new lines appeared during Tomcat startup:

![Screenshot from 2024-02-06 17-33-28](https://github.com/OneBusAway/onebusaway-docker/assets/856884/7cc31de5-46ae-45b0-a4bc-3daa137c5ef3)

aaronbrethorst committed:
Temporarily(?) disable the `build-bundle` service

aaronbrethorst committed:
Update the transit-data-federation-webapp data-sources.xml file

Update a handful of properties at the recommendation of an expert and at the recommendation of the OBA 2.x Config and Deploy Guide

aaronbrethorst committed:
Bump OBA binaries to v2.4.18

aaronbrethorst commented:
@ethanpooley I followed your recommendation on the bundle path. Oddly, I'm not seeing the console output you flagged. I also made a couple other changes to the `data-sources.xml` in question. thanks again for looking at this!

aaronbrethorst committed:
Removes API keys from deprecated clients

aaronbrethorst committed:
Fix DB connection

* Switch _back_ to MySQL
* Make sure the DB connection URLs in the data-sources.xml match the name of the DB server and DB.

aaronbrethorst committed:
Switch to the Unitrans GTFS feed

aaronbrethorst committed:
Remove the entrypoint script from mysql

Also, muck around with the (currently unused) environment variables for the OBA app server

aaronbrethorst committed:
Fix issues with loading arrivals-and-departures-for-stop endpoint
